### PR TITLE
Short circuit between DS and interface after DS downloads public key

### PR DESCRIPTION
**Pull request by utkarshgautamnyu**
With reference to fogbugz case https://highfidelity.fogbugz.com/f/cases/6547/Interface-waits-up-to-1s-before-re-attempting-connection-to-DS-after-DS-downloads-public-key\

This PR adds an additional short circuit between DS and interface which happens after the domain server has downloaded the public key for a user trying to connect

Testing Notes:
1) Run the built Domain Server and Interface
2) Assuming the user is already logged in, Observe the following in Domain Server logs:
    Extracted optimistic public key for "%username%"
    Username signature matches for "%username%"
3) Go to %APPDATA%/Roaming/High Fidelity{ - PRXXXX}/Interface/AccountInfo.bin and delete AccountInfo.bin
4) Restart Domain Server and Interface
5) Log in to High Fidelity in the interface
6) Observe the following logs in Domain Server:
    Extracted  public key for "%username%"
    Sending connection token packet to user "%username%" to trigger new check in
    Username signature matches for "%username%"
7) Observe Interface logs for the following: Received connection token packet from Domain Server...Re-sending check in

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2311/HighFidelity-Beta-PR11132-cc65f33a10634cb73cf49620ada195b909d0fb81.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2311/HighFidelity-Beta-PR11132-cc65f33a10634cb73cf49620ada195b909d0fb81.exe

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Passed
Test | Result
-------- | --------
Run the built Domain Server and Interface | Pass
**Assuming the user is already logged in, Observe the following in Domain Server logs**: | 
Extracted optimistic public key for "%username%" | Pass
Username signature matches for "%username%" | Pass
Go to %APPDATA%/Roaming/High Fidelity{ - PRXXXX}/Interface/AccountInfo.bin and delete AccountInfo.bin | Pass
Restart Domain Server and Interface | Pass
Log in to High Fidelity in the interface | See notes
**Observe the following logs in Domain Server**: | 
Extracted public key for "%username%" | Pass
Sending connection token packet to user "%username%" to trigger new check in | Pass
Username signature matches for "%username%" | Pass
Observe Interface logs for the following: Received connection token packet from Domain Server...Re-sending check in | Pass

## Test Notes

After logging into the client after deleting the AccountInfo.bin, my client crashed. I sent a bugsplat and re-followed the testing steps. Second run everything worked as expected without the crash. Not too sure what happened there. Might be worth looking into before merging.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: domain-server/src/DomainGatekeeper.cpp [hunk 1]
```diff
@@ -396,6 +396,7 @@ SharedNodePointer DomainGatekeeper::processAgentConnectRequest(const NodeConnect
             getGroupMemberships(username);
             verifiedUsername = username;
         } else {
+            _userSockAddress.insert(username , nodeConnection.senderSockAddr);
```
Comment (birarda) [inline domain-server/src/DomainGatekeeper.cpp]: there shouldn't be a space after `username` here

File: domain-server/src/DomainGatekeeper.cpp [hunk 2]
```diff
@@ -672,6 +674,13 @@ void DomainGatekeeper::publicKeyJSONCallback(QNetworkReply& requestReply) {
                 QByteArray::fromBase64(jsonObject[JSON_DATA_KEY].toObject()[JSON_PUBLIC_KEY_KEY].toString().toUtf8()),
                 isOptimisticKey
             };
+        
+        if (!isOptimisticKey) {
+            sendConnectionTokenPacket(username, _userSockAddress[username]);
+            _userSockAddress.remove(username);
+            qDebug().nospace() << "Sending connection token packet to user " << username << " to trigger new check in";
```
Comment (birarda) [inline domain-server/src/DomainGatekeeper.cpp]: you don't need the `nospace` here - remove space after `user` and before `to trigger` and remove the `.nospace()`

File: libraries/networking/src/NodeList.cpp [hunk 3]
```diff
@@ -575,6 +575,7 @@ void NodeList::processDomainServerConnectionTokenPacket(QSharedPointer<ReceivedM
     // read in the connection token from the packet, then send domain-server checkin
     _domainHandler.setConnectionToken(QUuid::fromRfc4122(message->readWithoutCopy(NUM_BYTES_RFC4122_UUID)));
     sendDomainServerCheckIn();
+    qDebug() << "Received connection token packet from Domain Server...Re-sending check in";
```
Comment (birarda) [inline libraries/networking/src/NodeList.cpp]: I think we should change the `...` to `,`

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2330/HighFidelity-Beta-PR11132-0e30ce49e83d0f410c729308897c809ccb7c987b.exe

**Comment by birarda** [comment]
Closing this since it needs a re-visit

**Pull request closed without merging**